### DISCUSSION TITLE
to.image "ValueError: 'png' is unsupported format"

### DISCUSSION THREAD

**Discussion by kristenwkristen** [jsvine/pdfplumber#933] [Ask for help with specific PDFs]
Hi, I am trying to use pdfplumber to convert an old record into data to move to excel. While trying to do visual debugging, I keep running into an issue with the to.image section of this code:

`import pdfplumber
ic35 = ('/Users/kristenwatkins/Desktop/corn/1935.pdf')

y = [55, 80, 110, 145, 180, 215, 245, 280, 315, 345, 490, 570] 
x = [135, 140, 147, 152, 157, 162, 167, 172, 179, 186, 192, 
     230, 235, 240, 246, 252, 258, 263, 268, 274, 280, 286, 292, 298, 304, 311, 317, 323, 329, 334, 343, 348, 354 
     ] 

with pdfplumber.open(ic35) as pdf:
    page = pdf.pages[11]
    im = page.to_image()
    im.reset().debug_tablefinder({
        "horizontal_strategy": "explicit",
        "vertical_strategy": "explicit",
        "explicit_vertical_lines": y,
        "explicit_horizontal_lines": x
    })
im`

This is the error I receive whenever I attempt this code and anytime I try to use .to_image. Any help would be appreciated as I've been stuck on this one error for days. 

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[1], line 19
     17 with pdfplumber.open(ic35) as pdf:
     18     page = pdf.pages[11]
---> 19     im = page.to_image()
     20     im.reset().debug_tablefinder({
     21         "horizontal_strategy": "explicit",
     22         "vertical_strategy": "explicit",
     23         "explicit_vertical_lines": y,
     24         "explicit_horizontal_lines": x
     25     })
     26 im

File ~/anaconda3/lib/python3.10/site-packages/pdfplumber/page.py:431, in Page.to_image(self, resolution, width, height)
    428 elif height is not None:
    429     resolution = 72 * height / self.height
--> 431 return PageImage(self, resolution=resolution or DEFAULT_RESOLUTION)

File ~/anaconda3/lib/python3.10/site-packages/pdfplumber/display.py:95, in PageImage.__init__(self, page, original, resolution)
     93 self.page = page
     94 if original is None:
---> 95     self.original = get_page_image(
     96         page.pdf.stream, page.page_number - 1, resolution
     97     )
     98 else:
     99     self.original = original

File ~/anaconda3/lib/python3.10/site-packages/pdfplumber/display.py:72, in get_page_image(stream, page_no, resolution)
     70 bg.composite(img, 0, 0)
     71 try:
---> 72     im = PIL.Image.open(BytesIO(bg.make_blob("png")))
     73 except PIL.Image.DecompressionBombError:
     74     raise PIL.Image.DecompressionBombError(
     75         "Image conversion raised a DecompressionBombError. "
     76         "PIL.Image.MAX_IMAGE_PIXELS is currently set to "
   (...)
     81         "#issuecomment-1190650404 for more information."
     82     )

File ~/anaconda3/lib/python3.10/site-packages/wand/image.py:9919, in Image.make_blob(self, format)
   9899 """Makes the binary string of the image.
   9900 
   9901 :param format: the image format to write e.g. ``'png'``, ``'jpeg'``.
   (...)
   9916 
   9917 """
   9918 if format is not None:
-> 9919     with self.convert(format) as converted:
   9920         return converted.make_blob()
   9921 library.MagickResetIterator(self.wand)

File ~/anaconda3/lib/python3.10/site-packages/wand/image.py:9802, in Image.convert(self, format)
   9782 """Converts the image format with the original image maintained.
   9783 It returns a converted image instance which is new. ::
   9784 
   (...)
   9799    and filename properties are aligned.
   9800 """
   9801 cloned = self.clone()
-> 9802 cloned.format = format
   9803 library.MagickSetFormat(cloned.wand,
   9804                         binary(format.strip().upper()))
   9805 return cloned

File ~/anaconda3/lib/python3.10/site-packages/wand/image.py:1882, in BaseImage.format(self, fmt)
   1880 r = library.MagickSetImageFormat(self.wand, binary(fmt.upper()))
   1881 if not r:
-> 1882     raise ValueError(repr(fmt) + ' is unsupported format')
   1883 r = library.MagickSetFilename(self.wand,
   1884                               b'buffer.' + binary(fmt.lower()))
   1885 if not r:  # pragma: no cover

ValueError: 'png' is unsupported format

**Comment by jsvine** [comment]
Hi @kristenwkristen, and strange; I've never seen that error before. My initial guess is that it might have something to do with an outdated/mismatched library. Could you do the following:

- Start a new Python virtual environment
- In that environment, run `pip install pdfplumber`
- Try your code again, this time in this virtual environment
- If it still fails, share the output of `pip freeze`

**Reply by kristenwkristen** [reply]
I think I have discovered that the problem is with ImageMagick and not within pdfplumber, but thank you for your help!

**Reply by jsvine** [reply]
Thanks for letting us know! 👍
